testsuite/tests/typing-poly/poly.ml

```diff
@@ -1098,7 +1098,7 @@ Line 2, characters 3-4:
 Error: This expression has type < m : 'a. 'a * < m : 'a * 'b > > as 'b
        but an expression was expected of type
          < m : 'a. 'a * (< m : 'a * < m : 'c. 'c * 'd > > as 'd) >
-       Types for method m are incompatible
+       The universal variable 'a would escape its scope
```

Comment:
Could we maybe have an `[%%expect]` block right after the `;;`, which is where the error arises if I understand correctly (on `fun`), and a different block for the type declarations that follow and `let f`?

The error message doesn't make much sense (before or after the change) to a human reader, but I understand why this is an error: if you unfold the recursive `< ... 'foo ...> as 'foo` types into infinite types, you can see that the two types are incompatible in their level of polymorphism (the same variable `'a` in the second type will occur infinitely many times, while it only occurs twice in the first one).